src/data/sym/basic.lean

```diff
@@ -150,4 +155,100 @@ instance inhabited_sym' [inhabited α] (n : ℕ) : inhabited (sym' α n) :=
 
 end inhabited
 
+instance has_zero : has_zero (sym α 0) := ⟨⟨0, rfl⟩⟩
+instance has_emptyc : has_emptyc (sym α 0) := ⟨0⟩
+
+instance subsingleton (n : ℕ) [subsingleton α] : subsingleton (sym α n) :=
+⟨begin
+  rintros ⟨a, ha⟩ ⟨b, hb⟩,
+  simp only [subtype.mk.inj_eq],
+  induction a using multiset.case_strong_induction_on with k hk ih generalizing n b,
+  { rw [ha.symm, multiset.card_zero] at hb, exact (multiset.card_eq_zero.mp hb).symm, },
+  { cases n,
+    { exact (multiset.cons_ne_zero (multiset.card_eq_zero.mp ha)).elim },
+    { by_cases hzero : b = 0,
+      { rw [hzero, multiset.card_zero] at hb,
+        exact (n.succ_ne_zero hb.symm).elim },
+      { have hmem := multiset.exists_mem_of_ne_zero hzero,
+        rcases hmem with ⟨r, hr⟩,
+        cases multiset.exists_cons_of_mem hr,
+        rw h,
+        have heq := @ih hk rfl.ge n w begin
+          rw multiset.card_cons at ha,
+          refine nat.succ.inj ha,
+        end begin
+          rw [h, multiset.card_cons] at hb,
+          refine nat.succ.inj hb,
+        end,
+        rw [heq, subsingleton.elim k r] } } }
+end⟩
+
+instance unique (n : ℕ) [unique α] : unique (sym α n) := unique.mk' _
+
+instance is_empty (n : ℕ) [is_empty α] : is_empty (sym α n.succ) :=
+⟨begin
+  intro h,
+  rw sym at h,
+  refine is_empty.exists_iff.mp (@multiset.exists_mem_of_ne_zero _ h.val _),
+  intro y,
+  have z := h.property,
+  rw [y, multiset.card_zero] at z,
+  exact (nat.succ_ne_zero n z.symm).elim,
+end⟩
```

Comment:
With the lemmas from my other comment, this can be golfed to
```lean
instance is_empty (n : ℕ) [is_empty α] : is_empty (sym α n.succ) :=
⟨λ s, by { obtain ⟨a, s, rfl⟩ := exists_eq_cons_of_succ s, exact is_empty_elim a }⟩
```